_studio/mfx_lib/shared/src/mfx_mpeg2_encode_vaapi.cpp

```diff
@@ -380,18 +387,74 @@ mfxStatus VAAPIEncoder::QueryEncodeCaps(ENCODE_CAPS & caps)
     caps.NoInterlacedField = 0; // Enable interlaced encoding
-    caps.SliceStructure    = 1; // 1 - SliceDividerSnb; 2 - SliceDividerHsw; 3 - SliceDividerBluRay; the other - SliceDividerOneSlice
 
+    std::map<VAConfigAttribType, size_t> idx_map;
+    VAConfigAttribType attr_types[] = {
+        VAConfigAttribMaxPictureHeight,
+        VAConfigAttribMaxPictureWidth,
+        VAConfigAttribEncSkipFrame,
+        VAConfigAttribEncMaxRefFrames,
+        VAConfigAttribEncSliceStructure
+    };
+
+    std::vector<VAConfigAttrib> attrs;
+    attrs.reserve(sizeof(attr_types) / sizeof(attr_types[0]));
+
+    for (size_t i=0; i < sizeof(attr_types)/sizeof(attr_types[0]); ++i)
+    {
+        attrs.push_back( createVAConfigAttr(attr_types[i], 0) );
+        idx_map[ attr_types[i] ] = i;
+    }
+
+    VAEntrypoint entrypoint = VAEntrypointEncSlice;
+
+    if(codecProfileType == 0)
+        codecProfileType = m_initCodecProfile;
+
+    VAStatus vaSts = vaGetConfigAttributes(m_vaDisplay,
+                          ConvertProfileTypeMFX2VAAPI(codecProfileType),
+                          entrypoint,
+                          attrs.data(), attrs.size());
+
+    MFX_CHECK(!(VA_STATUS_ERROR_UNSUPPORTED_ENTRYPOINT == vaSts ||
+                VA_STATUS_ERROR_UNSUPPORTED_PROFILE    == vaSts),
+                MFX_ERR_UNSUPPORTED);
+
+    if ((attrs[idx_map[VAConfigAttribMaxPictureWidth]].value != VA_ATTRIB_NOT_SUPPORTED) &&
+        (attrs[idx_map[VAConfigAttribMaxPictureWidth]].value != 0))
+        caps.MaxPicWidth  = attrs[idx_map[VAConfigAttribMaxPictureWidth]].value;
+    else
+        caps.MaxPicWidth = 1920;
+
+    if ((attrs[idx_map[VAConfigAttribMaxPictureHeight]].value != VA_ATTRIB_NOT_SUPPORTED) &&
+        (attrs[idx_map[VAConfigAttribMaxPictureHeight]].value != 0))
+        caps.MaxPicHeight = attrs[idx_map[VAConfigAttribMaxPictureHeight]].value;
+    else
+        caps.MaxPicHeight = 1920;
+
+    if ((attrs[idx_map[VAConfigAttribEncSkipFrame]].value != VA_ATTRIB_NOT_SUPPORTED) &&
+        (attrs[idx_map[VAConfigAttribEncSkipFrame]].value != 0))
+        caps.SkipFrame  = attrs[idx_map[VAConfigAttribEncSkipFrame]].value;
+    else
+        caps.SkipFrame = 1;
+
+    if ((attrs[idx_map[VAConfigAttribEncMaxRefFrames]].value != VA_ATTRIB_NOT_SUPPORTED) &&
+        (attrs[idx_map[VAConfigAttribEncMaxRefFrames]].value != 0))
+        caps.MaxNum_Reference  = attrs[idx_map[VAConfigAttribEncMaxRefFrames]].value;
+    else
+        caps.MaxNum_Reference = 1;
+
+    if ((attrs[idx_map[VAConfigAttribEncSliceStructure]].value != VA_ATTRIB_NOT_SUPPORTED) &&
+        (attrs[idx_map[VAConfigAttribEncSliceStructure]].value != 0))
+        caps.SliceStructure  = attrs[idx_map[VAConfigAttribEncSliceStructure]].value;
+    else
+        caps.SliceStructure = 2;
 
     return MFX_ERR_NONE;
-} // mfxStatus VAAPIEncoder::QueryEncodeCaps(ENCODE_CAPS & caps)
+} // mfxStatus VAAPIEncoder::QueryEncodeCaps(ENCODE_CAPS & caps, mfxU8 codecProfileType)
```

Comment:
just drop this comment entirely. It is useless.

Reply:
@dvrogozh all functions below have such a comment

Reply:
It's pretty useless comment..